### PR TITLE
Flip front face when world matrix determinant is negative

### PR DESCRIPTION
**Pull request by bghgary**
This is to support mirrored geometry as per glTF spec here: https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#transformations.

> Implementation Note: If the determinant of the transform is a negative value, the winding order of the mesh triangle faces should be reversed. This supports negative scales for mirroring geometry.

### CODE REVIEW

**Comment by deltakosh** [comment]
@sebavan do you mind reviewing this one as it touches PBR?

File: src/States/babylon.depthCullingState.ts [hunk 1]
```diff
@@ -38,19 +38,6 @@
             this._isZOffsetDirty = true;
         }
 
-        public get cullFace(): Nullable<number> {
```
Comment (deltakosh) [inline src/States/babylon.depthCullingState.ts]: I'm fine with this PR. My only concern is this change in the public API
We need to keep cullFace property (and probably deduce it from frontFace)

Comment (bghgary) [reply src/States/babylon.depthCullingState.ts]: I can put this back. Is it necessary to set it at all? I can just leave it as default.

Comment (deltakosh) [reply src/States/babylon.depthCullingState.ts]: it was as read/write property so wee need to keep it unaltered :D

**Comment by bghgary** [comment]
I have updated the PR with the changes.

**Comment by bghgary** [comment]
Updated again. In early versions, `engine.cullBackFaces` was incorrectly being ignored. This has been fixed in the current version. `cullFace` property on the `depthCullingState` object is actually required because of this.

I also included a one-line change to fix #3413.

**Pull request merged**